cmd/docker/docker.go

```diff
@@ -25,14 +26,17 @@ import (
 )
 
 func main() {
-	dockerCli, err := command.NewDockerCli()
+	ctx := context.Background()
+	dockerCli, err := command.NewDockerCli(command.WithBaseContext(ctx))
 	if err != nil {
 		fmt.Fprintln(os.Stderr, err)
 		os.Exit(1)
 	}
 	logrus.SetOutput(dockerCli.Err())
 
-	if err := runDocker(dockerCli); err != nil {
+	// TODO(krissetto): see if we can setup otel bits here, before
+	// setting up the cobra command by running dockerCli.Initialize() in `runDocker()`
```

Comment:
I don't mind the idea of wrapping the command too much, but honestly i'm not totally sure about it. 

My reasoning was that given how many special edge cases etc exist in this codebase, I would quite like for the otel init to happen "before everything else", so that we don't hamstring ourselves too much in terms of what we can measure end to end.

Regardless of where we do the init of the timer, on the cobra side what i would still do (and what i'm currently testing out a bit) is wrapping the commands with attributes, potentially another timer, etc